src/extension.ts

```diff
@@ -1,29 +1,50 @@
-import * as vsc from 'vscode';
-import * as openConsole from './commands/openConsole';
-import * as signIn from './commands/signIn';
-import * as signOut from './commands/signOut';
 import * as copyToClipboard from './commands/copyToClipboard';
-import * as selectMainProject from './commands/selectMainProject';
 import * as feelingLucky from './commands/feelingLucky';
 import * as fetchConfigs from './commands/fetchConfigs';
 import * as openConfigInConsole from './commands/openConfigInConsole';
+import * as openConsole from './commands/openConsole';
 import * as openTreeViewEntryInBrowser from './commands/openTreeViewEntryInBrowser';
-import UriHandler from './UriHandler';
-import ProjectsProvider from './providers/ProjectsProvider';
+import * as selectMainProject from './commands/selectMainProject';
+import * as signIn from './commands/signIn';
+import * as signOut from './commands/signOut';
+import * as toggleFavourite from './commands/toggleFavourite';
+import * as vsc from 'vscode';
+
+import {
+  ConfigCodeActionProvider,
+  registerCommands,
+} from './providers/ConfigCodeActionProvider';
+
 import AuthState from './state/AuthState';
-import ProjectsState from './state/ProjectsState';
+import ConfigCodeLensProvider from './providers/ConfigCodeLensProvider';
 import ConfigHoverProvider from './providers/ConfigHoverProvider';
+import ProjectsProvider from './providers/ProjectsProvider';
+import ProjectsState from './state/ProjectsState';
+import UriHandler from './UriHandler';
 import getExtensionConfig from './state/getExtensionConfig';
-import ConfigCodeLensProvider from './providers/ConfigCodeLensProvider';
 import { subscribeToDocumentChanges } from './providers/diagnostics';
-import {
-  registerCommands,
-  ConfigCodeActionProvider,
-} from './providers/ConfigCodeActionProvider';
 
 export function activate(context: vsc.ExtensionContext): void {
+  const globalState: vsc.Memento = context.globalState;
+
+  const getData = (key: string): boolean => {
+    return key ? globalState.get(key) ?? false : false;
```

Comment:
You can just do `globalState.get(key) ?? false` without checking the key first. If the key is null / invalid / doesn't exist, it'll goto false anyways

Reply:
ahh okok